### ISSUE TITLE
Markdown: Bold/italic syntax in code misinterpreted as opening/closing formatting blocks

### ISSUE THREAD

**Issue by queengooborg** [prettier/prettier#15032]
When defining a bold/italic span, the syntax inside of inline code is misinterpreted as opening or closing of the formatting span.  This causes Prettier to escape the real opening/closing syntax.

**Prettier 2.8.8** (also occurs in 3.0.0 locally)
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEBGJACAKgCzhgKgLgA8AHaBGASwEMboMAKAAyJYEoiMIy4AnehH4A6ADpQJAJky58xcpVh0GUZm04EefQTGHjJUAMyy8GAPqkKUKiuqNW5zue0ChoiRIAsp-OcuKNsr09mqOThz+rrr6IAA0ILyqAM7IoLT8-BAA7gAKGQipKLQANtm0AJ6pCQBGgmAA1nAwAMq0ALZwADLUNsgAZqXJcLX1Ta1ktGC9AObIMPwAriMgcO01cAAmm1tdtFAzi7QzcABiwu30NAfIILSLevEgODDtJQDqONTwyZNgcC1Ct9qAA3b4VW5gZLVEC9Yb8GC5QQzS4DIYrABWyRILVmJTgAEVFhB4GiSsMEpN+PDbpd+A1NjkoE8yPxejB3tRNjAcMgABwABkpWWG70EZFurLg8JBcCeAEdifAkbwindkgBaGxbLZPfhwRXUfVI46opCDckrYbtajzJZWvFwACCMAW1BqDzguQEPT65vRCWSjqJJLl-stCRgtBqnO5vKQUkjgmoJVmAGEIO0zatkgBWJ6LYZYaNFC0UkAg5YASSgO1gLTAbLIMCdtZaMAq+LJwwAvj2gA)

```sh
--parser markdown
```

**Input:**

```markdown
1: The **exponentiation (`**`)** operator.

2: The *exponentiation (`*`)* operator.

3: The _exponentiation (`_`)_ operator.

4: The __exponentiation (`__`)__ operator.
```

**Output:**

```markdown
1: The **exponentiation (`**`)\*\* operator.

2: The _exponentiation (`_`)\* operator.

3: The _exponentiation (`_`)\_ operator.

4: The **exponentiation (`**`)\_\_ operator.

```

**Expected behavior:**

```markdown
1: The **exponentiation (`**`)** operator.

2: The _exponentiation (`*`)_ operator.

3: The _exponentiation (`_`)_ operator.

4: The **exponentiation (`__`)** operator.
```

**Comment by sosukesuzuki** [comment]
I think there was another issue that was similar, but I forget...

**Comment by auvred** [comment]
This bug caused by `remark-parse@8` (it produces incorrect AST, [playground](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEB9AhgA1dgOlEAGhAgAcYBLaAZ2VHQCcGIB3ABUYVpXQBsX0AT1rEARg3RgA1nBgBldAFs4AGQpQ4yAGZ9qcMROmy5pSeoDmyGAwCu+kHsUUrt+9Qu84ARRsR423XsAK2oADzkPb19-JB1ePWIAR2i4NmZSbhB0agBaDTgAE0KiEGt0Cl4LAGEIRUV0ZBA4agBWEvcoc08AQRhrClEbeDY4BjUNAPj7AAsYRV4AdWmKeGpTMDg5LhWKADcVwUawahEQXbsASSgi2DkwBgpybuu5GEFPSYSQUmY9BYlSI0fs1RrtNMR1HoGDA0uhzPVPvZTAwoY16gwpAVWARiD91DAFhQCjBpsgABwABmIDDgyQoNNh8IasUCxBg6FEhOJpKQACZiDY9AAVDncOJfOCKUSFIoFFToTo2OFwABiEAY9T6Fka6CGEBAAF8DUA)). Updating it to 9+ should help. The issue is tracker here: https://github.com/prettier/prettier/issues/11828

**Comment by asvishnyakov** [comment]
2026 and the bug is still there

**Comment by tats-u** [comment]
Probably already fixed by https://github.com/prettier/prettier/pull/18277 (select `next` in playground). Probably fixed by the next minor version.

https://deploy-preview-18277--prettier.netlify.app/playground/#N4Igxg9gdgLgprEAuEBGJACAKgCzhgKgLgA8AHaBGASwEMboMAKAAyJYEoiMIy4AnehH4A6ADpQJAJky58xcpVh0GUZm04EefQTGHjJUAMyy8GAPqkKUKiuqNW5zue0ChoiRIAsp-OcuKNsr09mqOThz+rrr6IAA0ILyqAM7IoLT8-BAA7gAKGQipKLQANtm0AJ6pCQBGgmAA1nAwAMq0ALZwADLUNsgAZqXJcAkQNQBWcGAwAOqCZMggZPxww-wAbnDxIHW0jc0tZHu9AObIMPwAriMgw+3U51c3VgLUnbClAPI67rkQydRVIsEAATbYvfhvKilLACKAZairAZDG4AqAnEpwACKlwg8GRJWGCXGyRILVOmJxeK2SEGhJuAEdcfBclkyEUQLRkgBaGxwEH87YXWjUEqnADCEHa7Voi1KJW2aIxcAAgjALtQapcWQIen1aSiEjgYO0SjMcIDVkcwHAWoVAdR1oCKoswMlqiB1tcAJJQAWwFpgSFkGAqv0tGAVTEEolLLLDOa0BYoZarASbcF+z79PU0kAlfrbXprGCs2gnGUxm5HfhrRYy-gNEE5KDbZa9WbUEEwHDIAAcAAYEismdQVmWK7KDfSEjBaDUZl2e8gpAlLsMsPOinTY3B2jV+QKQV1aOjLuW4AAxYQy9WnOXaiAgAC+z6AA

**Comment by pszypowicz** [comment]
Quick FYI for anyone testing this: the fix is on `main`, but it isn't in any released npm artifact yet, including `prettier@next`.

Tested the canonical repro from this issue (`` The **exponentiation (`**`)** operator. ``) against three versions:

- `prettier@latest` (3.8.3): bug present.
- `prettier@next` (4.0.0-alpha.13, published 2025-11-18): bug present - this alpha pre-dates the #18277 merge (2025-12-26).
- Built from `main` (3.9.0-dev, HEAD today): output identical to input. Fixed.

So `npm install prettier@next` won't show the fix today; building from source or waiting on 3.9.0 are the only paths.
